config/routes.rb

```diff
@@ -1,6 +1,12 @@
 Rails.application.routes.draw do
-  resources :repositories, only: %i[index new create]
-  root 'repositories#index'
+  root 'static_pages#index'
+
+  scope module: 'author_admin', path: 'author', as: 'author_admin' do
+    resources :repositories do
+      post 'rebuild', on: :member
```

Comment:
create another entity likes 'repositories' to support this 'post', e.g. 'building:create'